### PR TITLE
[MBL-17778][Student] - Show 'No Internet Connection' dialog when click on Anonymous discussion details 'Open in browser' link

### PR DESCRIPTION
**Pull request by kdeakinstructure**
Show 'No Internet Connection' dialog when click on 'Open in browser' link in offline mode, on anonymous discussion details screen.

Test plan: Check that click on 'Open in browser' link in an anonymous discussion details page in offline mode will show the 'No Internet Connection' dialog.

refs: MBL-17778
affects: Student
release note: none

- [x] Tested in dark mode
- [x] Tested in light mode

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/student/src/main/java/com/instructure/student/features/discussion/details/DiscussionDetailsFragment.kt [hunk 1]
```diff
@@ -634,12 +676,16 @@ class DiscussionDetailsFragment : ParentFragment(), Bookmarkable {
         replyToDiscussionTopic.setGone()
         swipeRefreshLayout.isEnabled = false
         openInBrowser.onClick {
```
Comment (kristofnemere) [inline apps/student/src/main/java/com/instructure/student/features/discussion/details/DiscussionDetailsFragment.kt:678]: we have an extension for this: onClickWithRequireNetwork

**Review by kristofnemere — CHANGES_REQUESTED** [review]

File: apps/student/src/main/java/com/instructure/student/features/discussion/details/DiscussionDetailsFragment.kt (hunk 1 above)
Comment (kdeakinstructure) [reply apps/student/src/main/java/com/instructure/student/features/discussion/details/DiscussionDetailsFragment.kt]: Thanks, done

**Review by kristofnemere — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — APPROVED** [review]

**Review by hermannakos — APPROVED** [review]

**Pull request merged**